SteamKit2/SteamKit2/Networking/Steam3/TcpConnection.cs

```diff
@@ -187,10 +188,14 @@ public override void Connect(IPEndPoint endPoint, int timeout)
                     Debug.Assert(cancellationToken == null);
                     cancellationToken = new CancellationTokenSource();
 
-                    destination = endPoint;
                     socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
 
-                    ThreadPool.QueueUserWorkItem(TryConnect, timeout);
+                    endPoint.ContinueWith(t =>

```

Comment:
What if `endPoint` (suggest renaming to `endPointTask`) is faulted or cancelled? IIRC `t.Result` will blow up, on some thread we don't control.